driver/attributes.h

```diff
@@ -45,80 +44,85 @@ inline bool AttributeContainer::hasAttrAs<std::string>(int attr) const {
 
 template <typename T>
 inline T AttributeContainer::getAttrAs(int attr, const T & def) const {
-    auto it = integers.find(attr);
-    return (it == integers.end() ?
-        def : (T)(it->second));
+    const auto it = attributes.find(attr);
+
+    if (it == attributes.end() || it->second.valueless_by_exception())
+        return def;
+
+    return std::visit([] (auto & value) -> T {
+        using ValueType = std::decay_t<decltype(value)>;
+
+        if constexpr (std::is_same_v<ValueType, std::string>) {
```

Comment:
Why not just something this simple ?
```
if constexpr (is_same_v<ValueType, T>) {
    return value;
}
```

And why do you need this function no cast attribute value from string to int ?

Reply:
To avoid built-in edge cases. Attributes either behave intuitively (convert any to any, if possible, or fail), or fail to compile. But never silently return default values, on failure, etc.

Reply:
well, I guess we need more tests for that. And negative tests too.